WordPress/src/main/java/org/wordpress/android/ui/sitecreation/verticals/SiteCreationIntentsFragment.kt

```diff
@@ -78,11 +82,28 @@ class SiteCreationIntentsFragment : Fragment() {
     private fun SiteCreationIntentsFragmentBinding.updateUiState(uiState: IntentsUiState) {
         (recyclerView.adapter as SiteCreationIntentsAdapter).update(uiState.content.items)
         updateTitleVisibility(uiState.isAppBarTitleVisible)
-        if (!uiState.isHeaderVisible) input.requestFocus()
+
+        uiState.vertical
+                ?.let {
+                    input.requestFocus()
+                    ActivityUtils.showKeyboard(input)
+                }
+                ?: run {
```

Comment:
I have a slight concern about the behavior if `showKeyboard` ever returns `null`, since I guess this will also invoke the chained run block :thinking:

Reply:
That's a valid remark @mkevins! Thank you for sharing it 🙇‍♂️.

I agree we don't want changes to the return value of `ActivityUtils.showKeyboard` to influence this behavior 👍.

To make the code safer I wrapped the calls into an `also` block which makes sure to return the value of `uiState.vertical` in https://github.com/wordpress-mobile/WordPress-Android/pull/16380/commits/ae0528c57df33ae3f6bd024df232c8b900dc7253.